### PR TITLE
✨Source Public-Api : Migrate Python CDK to Low-code CDK

### LINKED ISSUE #29454: public api
assigned to @avirajsingh7 

https://github.com/airbytehq/airbyte/pull/29391

### PR DESCRIPTION
**Pull request by avirajsingh7**
## What
Migrating Source public-api to Low-Code CDK
Closes #29454

## How
Developed using (Configuration Based Source) low-code CDK

## Recommended reading order

1. spec.yaml
2. manifest.yaml
3. schemas/*

##  Test
<details>
<summary>Acceptance Test</summary>

There is no authentication still one of the auth test fails

Screenshot 2023-08-13 at 9 06 40 PM
</details>
<details>
<summary>Migrate Connector : Source Public-api</summary>

- [x] **Community Member? Grant Edit Access to Maintainers [instructions](https://docs.github.com/en/pull-requests/collaborating-with-pull-requests/working-with-forks/allowing-changes-to-a-pull-request-branch-created-from-a-fork#enabling-repository-maintainer-permissions-on-existing-pull-requests)**
- [x] **Secrets in the Connector's Spec**
- [x] **Unit & integration tests added and passing. Community members, please provide proof of success locally e.g: screenshot or copy-paste unit, integration, and acceptance test output. To run acceptance tests for a Python connector, follow instructions in the README. For java connectors run ./gradlew :airbyte-integrations:connectors:<name>:integrationTest.**
- [x] **Code Reviews Completed**
  - [x] **Documentation Updated**
  - [x] **Connector's README.md**
  - [x] **Connector's bootstrap.md**
  - [x] **docs/integrations/<source or destination>/<name>.md (including changelog. See changelog example)**
  - [x] **docs/integrations/README.md**
  - [x] **airbyte-integrations/builds.md**
- [x] **PR Name follows PR Naming Conventions**
</details>

### CODE REVIEW

File: airbyte-integrations/connectors/source-public-apis/source_public_apis/components.py [hunk 1]
```diff
@@ -0,0 +1,15 @@
...
+from airbyte_cdk.sources.declarative.extractors.record_extractor import RecordExtractor
+from typing import Any, List
+import requests
+
+class CustomExtractor(RecordExtractor):
+    
+    def extract_records(self, response: requests.Response, **kwargs) -> List[Mapping[str, Any]]:
+
+        extracted=[]
+        print(response.json())
```
Comment (marcosmarxm) [inline airbyte-integrations/connectors/source-public-apis/source_public_apis/components.py:12]: Remove the print.

File: airbyte-integrations/connectors/source-public-apis/source_public_apis/components.py [hunk 2]
```diff
@@ -0,0 +1,15 @@
...
+        for cat in response.json()['categories']:
+            extracted.append({"name": cat})
+        return extracted
```
Comment (marcosmarxm) [inline airbyte-integrations/connectors/source-public-apis/source_public_apis/components.py:15]: Can you keep the same function as before `        return [{"name": cat} for cat in response.json()["categories"]]
`

**Review by marcosmarxm — CHANGES_REQUESTED** [review]
Overall looks good. Please update the doc entry too.

File: airbyte-integrations/connectors/source-public-apis/source_public_apis/components.py (hunk 2 above)
Comment (avirajsingh7) [reply airbyte-integrations/connectors/source-public-apis/source_public_apis/components.py]: resolved

**Comment by avirajsingh7** [comment]
@marcosmarxm Done suggested changes

**Review by marcosmarxm — APPROVED** [review]

**Comment by marcosmarxm** [comment]
Published in https://github.com/airbytehq/airbyte/pull/29951

**Pull request closed without merging**